File: Data Structures & Algorithms/pascals-triangle/submission-0.py
class Solution:
    def generate(self, n: int) -> List[List[int]]:
        ans = []
        for i in range(n):
            arr = []
            for j in range(i+1):
                if j == 0 or j == i:
                    arr.append(1)
                elif i > 0:
                    arr.append(ans[i-1][j] + ans[i-1][j-1])
            ans.append(arr)
        return ans

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string ans = "";
        for (auto s: strs) {
            int n = s.size();
            ans += to_string(n) + "#" + s;
        }
        return ans;
    }

    vector<string> decode(string str) {
        vector<string> ans;
        int i = 0;
        while (i < str.size()) {
            int sz = 0;
            while (str[i] != '#') {
                sz *= 10;
                sz += str[i]-'0';
                i++;
            }
            i++;
            string s = str.substr(i, sz);
            ans.push_back(s);
            i += sz;
        }
        return ans;
    }
};
